packages/DataGrid/columns/number/DataGridColumnCurrency.ts

```diff
@@ -13,16 +13,24 @@ export class DataGridColumnCurrency<TData> extends DataGridColumnNumberBase<TDat
 	static defaultCurrency?: Currency
 
 	@property({ type: Object, converter: FieldCurrency.currencyConverter }) currency = DataGridColumnCurrency.defaultCurrency
+	@property() currencyDataSelector?: string
 
-	// eslint-disable-next-line @typescript-eslint/no-unused-vars
-	getContentTemplate(value: number | undefined, _data: TData) {
-		return html`${this.getNumber(value)?.formatAsCurrency(this.currency!) ?? html.nothing}`
+	getContentTemplate(value: number | undefined, data: TData) {
+		const currency = this.currencyDataSelector
+			? Currency[(data as any)[this.currencyDataSelector] as CurrencyCode]
+			: this.currency
+
+		return html`${this.getNumber(value)?.formatAsCurrency(currency!) ?? html.nothing}`
 	}
 
 	getEditContentTemplate(value: number | undefined, data: TData) {
+		const currency = this.currencyDataSelector
+			? Currency[(data as any)[this.currencyDataSelector] as CurrencyCode]
+			: this.currency
```

Comment:
Let's not repeat this logic and outsource it to a getter accessor called `private _currency()` or similar